homeassistant/components/bthome/event.py

```diff
@@ -0,0 +1,136 @@
+"""Support for bthome event entities."""
+from __future__ import annotations
+
+from dataclasses import replace
+
+from homeassistant.components.event import (
+    EventDeviceClass,
+    EventEntity,
+    EventEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import format_discovered_event_class, format_event_dispatcher_name
+from .const import (
+    DOMAIN,
+    EVENT_CLASS_BUTTON,
+    EVENT_CLASS_DIMMER,
+    EVENT_PROPERTIES,
+    EVENT_TYPE,
+    BTHomeBleEvent,
+)
+from .coordinator import BTHomePassiveBluetoothProcessorCoordinator
+
+DESCRIPTIONS_BY_EVENT_CLASS = {
+    EVENT_CLASS_BUTTON: EventEntityDescription(
+        key=EVENT_CLASS_BUTTON,
+        translation_key="button",
+        event_types=[
+            "press",
+            "double_press",
+            "triple_press",
+            "long_press",
+            "long_double_press",
+            "long_triple_press",
+        ],
+        device_class=EventDeviceClass.BUTTON,
+    ),
+    EVENT_CLASS_DIMMER: EventEntityDescription(
+        key=EVENT_CLASS_DIMMER,
+        translation_key="dimmer",
+        event_types=["rotate_left", "rotate_right"],
+    ),
+}
+
+
+class BTHomeEventEntity(EventEntity):
+    """Representation of a BTHome event entity."""
+
+    _attr_should_poll = False
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        address: str,
+        event_class: str,
+        event: BTHomeBleEvent | None,
+    ) -> None:
+        """Initialise a BTHome event entity."""
+        self._update_signal = format_event_dispatcher_name(address, event_class)
+        # event_class is something like "button" or "dimmer"
+        # and it maybe postfixed with "_1", "_2", etc
+        base_event_class, _, postfix = event_class.partition("_")
+        base_description = DESCRIPTIONS_BY_EVENT_CLASS[base_event_class]
+        self.entity_description = replace(base_description, key=event_class)
+        postfix_one_indexed = int(postfix) + 1 if postfix else None
+        postfix_name = f" {postfix_one_indexed}" if postfix else ""
+        # We report button button 1 as button 2, etc to the user since
```

Comment:
This isn't needed, the bthome-ble package already handles this. 

Devices with one button event won't get a postfix (button)
Devices with e.g. three button events will get a postfix, also the first (`button_1`, `button_2` and `button_3`)

if you have e.g. `b"\x40\x3A\x00"`, the package won't add a postfix --> device with one button, no postfix needed.
if you have e.g. `b"\x40\x3A\x01\x3A\x01\x3A\x03"`, the package will add a postfix to all three buttons, including the first (`button_1`, `button_2` and `button_3`. 
if you have e.g. `b"\x40\x3A\x00\x3A\x01\x3A\x03"`, like in the test, the package will skip the first button (`0x00` = no event), and use postfix 2 and 3 for the 2nd and 3rd button (`button_2` and `button_3`). 

So, I don't see the need to add 1 to the postfix. 

Reference:
Devices with multiple buttons are identified [here](https://github.com/Bluetooth-Devices/bthome-ble/blob/3bf6e69bf3c6f81bcd28ae7bbb7119be9e7b4876/src/bthome_ble/parser.py#L465)
The postfix is added [here](https://github.com/Bluetooth-Devices/bthome-ble/blob/3bf6e69bf3c6f81bcd28ae7bbb7119be9e7b4876/src/bthome_ble/parser.py#L483), so, only for devices with multiple buttons, and also for the first button.

Reply:
I figured it was zero indexed.. adjusted in 58ecf7ac33b7acacd8275fa7d61877b441a7540a